engine/src/flutter/lib/ui/painting.dart

```diff
@@ -2292,6 +2292,16 @@ class FrameInfo {
   final Image image;
 }
 
+/// Signature for [Codec] lifecycle events.
+typedef CodecEventCallback = void Function(Codec codec);
+
+/// If true, Codec is leak tracked.
+///
+/// For use in Flutter tests only.
+/// Do not take dependency. It will be deleted without notice.
+// TODO (https://github.com/flutter/flutter/issues/161132): remove after fixing the tests.
+bool deprecatedDoNotUseWillBeRemovedWithoutNoticeLeakTrackCodec = false;
```

Comment:
I see a couple of issues with this:

1. It introduces a public API to the `dart:ui` API surface. I realize that it has a scary name, but this will make it into our public documentation, and even accidental dependencies are difficult to get rid of. Generally we don't use this approach. Is there a different way to achieve the same thing? For example, now that both the engine and the framework are in one monorepo, can the failing tests be fixed right now in this PR?
2. This is only added to the mobile version of `dart:ui`. An identical change must be made in the web version. Otherwise, the dart2js/dart2wasm won't be able to compile the framework-side code.

Reply:
Thanks! Yes, we can add this flag to other place.